python/ray/data/datasource/datasource.py

```diff
@@ -217,10 +217,7 @@ def __call__(self) -> Iterable[Block]:
             for block in result:
                 yield from self._do_additional_splits(block)
         else:
-            builder = DelegatingBlockBuilder()
-            for block in result:
-                builder.add_block(block)
-            yield builder.build()
+            yield from result
```

Comment:
Isn't this path only taken if block splitting is disabled? It should be on by default

Reply:
> Isn't this path only taken if block splitting is disabled? 

That's right. I had disabled it because I ran into a bug with block splitting.